### PR TITLE
Add a check to verify RGW endpoint is reachable or not

### PR DESCRIPTION
**Pull request by aruniiird**
Added an extra check to verify RGW endpoint passed,
in External cluster mode, is reachable or not.

Signed-off-by: Arun Kumar Mohan <amohan@redhat.com>

### CODE REVIEW

**Comment by aruniiird** [comment]
@jarrpa , @obnoxxx , @umangachapagain , please take a look...

**Comment by obnoxxx** [comment]
Thanks @aruniiird for the PR!
As discusses offline, this needs a little more discussion and investigation - the correct place for the fix might be in rook.

/hold

FYI @travisn @leseb @jarrpa

File: pkg/controller/storagecluster/initialization_reconciler.go [hunk 1]
```diff
@@ -196,7 +207,13 @@ func (r *ReconcileStorageCluster) ensureExternalStorageClusterResources(instance
 			} else if d.Name == cephRgwStorageClassName {
 				// Set the external rgw endpoint variable for later use on the Noobaa CR (as a label)
 				// Replace the colon with an underscore, otherwise the label will be invalid
-				externalRgwEndpointReplaceColon := strings.Replace(d.Data[externalCephRgwEndpointKey], ":", "_", -1)
+				rgwEndpoint := d.Data[externalCephRgwEndpointKey]
+				if err := checkRGWEndpoint(rgwEndpoint, 30*time.Second); err != nil {
```
Comment (leseb) [inline pkg/controller/storagecluster/initialization_reconciler.go:211]:
```suggestion
				if err := checkRGWEndpoint(rgwEndpoint, 5*time.Second); err != nil {
```

A lower 5s timeout should be more than enough to validate the connectivity.

Comment (aruniiird) [reply pkg/controller/storagecluster/initialization_reconciler.go]: Done

**Comment by aruniiird** [comment]
I have edited the commit message, so that it reflects an additional / required check rather than bug fix.
@jarrpa , @obnoxxx  please take a look...

**Comment by jarrpa** [comment]
Please investigate the unit test failures.

**Comment by aruniiird** [comment]
@jarrpa , corrected the testcases. Please take a look...

File: pkg/controller/storagecluster/external_resources_test.go [hunk 2]
```diff
@@ -55,7 +57,8 @@ var ExternalResources = []ExternalResource{
 	ExternalResource{
 		Kind: "StorageClass",
 		Data: map[string]string{
-			"endpoint": "10.20.30.40:50",
+			// "endpoint": "10.20.30.40:50",
```
Comment (jarrpa) [inline pkg/controller/storagecluster/external_resources_test.go:60]: Remove this comment

File: pkg/controller/storagecluster/external_resources_test.go [hunk 3]
```diff
@@ -349,3 +355,34 @@ func assertCephObjectStore(t *testing.T, reconciler ReconcileStorageCluster, rem
...
+
+func generateRandomPort(minPort, maxPort int) int {
+	rand.Seed(time.Now().UnixNano())
+	portRange := minPort - maxPort
+	if portRange < 0 {
+		portRange *= -1
+	}
+	retPort := rand.Intn(portRange) + minPort
+	return retPort
+}
```
Comment (jarrpa) [inline pkg/controller/storagecluster/external_resources_test.go:388]: Move these above the Test function

File: pkg/controller/storagecluster/external_resources_test.go (hunk 2 above)
Comment (aruniiird) [reply pkg/controller/storagecluster/external_resources_test.go]: Done...

File: pkg/controller/storagecluster/external_resources_test.go (hunk 3 above)
Comment (aruniiird) [reply pkg/controller/storagecluster/external_resources_test.go]: Moved it along side with other helper functions in the file...

**Pull request merged**